### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by MaksimZhukov**
In scope of this PR we added tests. Also, we added workflow file to release new action version

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@master
```
Comment (maxim-lobanov) [inline .github/workflows/test.yml:13]: `actions/checkout@v2`

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    - name: Set Node.JS
+      uses: actions/setup-node@master
+      with:
+        node-version: 12.x
+
+    - name: npm install
```
Comment (AlenaSviridenko) [inline .github/workflows/test.yml:20]: For consistency, I'd suggest rename to "Install dependencies"

File: __tests__/api-utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import * as github from '@actions/github';
+import * as apiUtils from "../src/api-utils";
+
+const prereleaseData = require('./data/pre-release.json');
+const releaseData = require('./data/release.json');
+
+const token = "faketoken";
```
Comment (AlenaSviridenko) [inline __tests__/api-utils.test.ts:7]: quotes, again, they are different everywhere

File: __tests__/version-utils.test.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        expect(() => versionUtils.validateSemverVersionFromTag("v1.0.0+20130313144700")).not.toThrow();
+    });
+
+    it("throw when a tag contains an invalid semantic version", () => {
+        expect(() => versionUtils.validateSemverVersionFromTag("1.0.0invalid")).toThrowError(
+            "The '1.0.0invalid' doesn't satisfy semantic versioning specification"
+        );
+    });
+
+    describe("throw when a tag contains an valid unstable semantic version", () => {
```
Comment (AlenaSviridenko) [inline __tests__/version-utils.test.ts:22]: why do you put into a separate describe?

File: __tests__/api-utils.test.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+        expect.assertions(1);
+        await expect(apiUtils.validateIfReleaseIsPublished("v1.1.0", octokitClient)).resolves.not.toThrow();
+    });
+
+    afterEach(() => {
+        jest.resetAllMocks();
+        jest.clearAllMocks();
+    });
+});
```
Comment (AlenaSviridenko) [inline __tests__/api-utils.test.ts:39]: is it possible to add test for tag update? Maybe with some mocks

**Review by vsafonkin — APPROVED** [review]

File: __tests__/version-utils.test.ts (hunk 4 above)
Comment (MaksimZhukov) [reply __tests__/version-utils.test.ts:22]: Just to make logs more readable:
image

File: __tests__/api-utils.test.ts (hunk 5 above)
Comment (MaksimZhukov) [reply __tests__/api-utils.test.ts:39]: The `updateTag` function consists of several API calls and one `if else` condition. In my understanding, it doesn't make sense to mock all of them to test one condition, does it?

File: .github/workflows/test.yml (hunk 1 above)
Comment (MaksimZhukov) [reply .github/workflows/test.yml]: Thanks!

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@master
+
+    - name: Set Node.JS
+      uses: actions/setup-node@master
```
[the remark this answers is not in the document]
Comment (MaksimZhukov) [reply .github/workflows/test.yml]: Thanks!

File: .github/workflows/test.yml (hunk 2 above)
Comment (MaksimZhukov) [reply .github/workflows/test.yml]: Thanks!

File: __tests__/api-utils.test.ts (hunk 3 above)
Comment (MaksimZhukov) [reply __tests__/api-utils.test.ts:7]: ohh, I see, thanks

File: __tests__/data/pre-release.json [hunk 7]
```diff
@@ -0,0 +1,77 @@
+{
```
[the remark this answers is not in the document]
Comment (MaksimZhukov) [reply __tests__/data/pre-release.json:1]: Thanks!

File: __tests__/api-utils.test.ts (hunk 5 above)
Comment (AlenaSviridenko) [reply __tests__/api-utils.test.ts:39]: I stick to the idea of covering everything with tests, but if you think it is overkill, then we can live without this test

**Review by AlenaSviridenko — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Pull request merged**